lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py

```diff
@@ -61,89 +61,248 @@
+            - Inbound NAT rules are created automatically for each NIC associated with the Load Balancer using an external port from this range.
+            - Defining an Inbound NAT pool on your Load Balancer is mutually exclusive with defining inbound Nat rules.
+            - Inbound NAT pools are referenced from virtual machine scale sets.
+            - NICs that are associated with individual virtual machines cannot reference an inbound NAT pool.
+            - They have to reference individual inbound NAT rules.
+        suboptions:
+            name:
+                description: name of the inbound nat pool.
+                required: True
+            frontend_ip_configuration_name:
+                description: A reference to frontend IP addresses.
+                required: True
+            protocol:
+                description: "Possible values include: 'Udp', 'Tcp', 'All'."
+                choices:
+                    - Tcp
+                    - Udp
+                    - All
+                default: Tcp
+            frontend_port_range_start:
+                description:
+                    - The first port number in the range of external ports that will be used to provide Inbound Nat to NICs associated with a load balancer.
+                    - Acceptable values range between 1 and 65534.
+                required: True
+            frontend_port_range_end:
+                description:
+                    - The last port number in the range of external ports that will be used to provide Inbound Nat to NICs associated with a load balancer.
+                    - Acceptable values range between 1 and 65535.
+                required: True
+            backend_port:
+                description:
+                    - The port used for internal connections on the endpoint.
+                    - Acceptable values are between 1 and 65535.
+        version_added: 2.5
+    load_balancing_rules:
+        description:
+            - Object collection representing the load balancing rules Gets the provisioning.
+        suboptions:
+            name:
+                description: name of the load balancing rule.
+                required: True
+            frontend_ip_configuration:
+                description: A reference to frontend IP addresses.
+                required: True
+            backend_address_pool:
+                description: A reference to a pool of DIPs. Inbound traffic is randomly load balanced across IPs in the backend IPs.
+                required: True
+            probe:
+                description: The reference of the load balancer probe used by the load balancing rule.
+                required: True
+            protocol:
+                description: "Possible values include: 'Udp', 'Tcp', 'All'."
+                choices:
+                    - Tcp
+                    - Udp
+                    - All
```

Comment:
also the recent rule is to have choice values in snake_lower_case format

Reply:
These are strings accepted by SDK.